### PR TITLE
14.0.0

### PR DESCRIPTION
**Pull request by adonesky1**
This is the release candidate for version 14.0.0.

### Added
- **BREAKING** Add EIP1559 support including `speedUpTransaction` and `stopTransaction` ([#521](https://github.com/MetaMask/controllers/pull/521))
 - The breaking change here is that consumers of this repo now have to check if the transaction object includes a gas price and fetch and add it themselves (if need be).
### Changed
- Make equality comparisons for token and collectible addresses in TokensController and CollectiblesController case insensitive  ([#537](https://github.com/MetaMask/controllers/pull/537))
- Reset gas fee estimate polling onNetworkStateChange in the gasFeeController ([#534](https://github.com/MetaMask/controllers/pull/534))

### Fixed
- Update AssetDetectionController to handle new fetch limits for the OpenSea collectibles api ([#536](https://github.com/MetaMask/controllers/pull/536))

### CODE REVIEW

File: CHANGELOG.md
Comment (Gudahtt) [inline CHANGELOG.md:11]: No PR link? This was #521

```suggestion
- Add EIP1559 support including `speedUpTransaction` and `stopTransaction` ([#521](https://github.com/MetaMask/controllers/pull/521))
```

**Review by Gudahtt — COMMENTED** [review]
#534 and #537 seem worthy of change entries as well. They were merged after this was created I think.

**Comment by adonesky1** [comment]
> #534 and #537 seem worthy of change entries as well. They were merged after this was created I think.

addressed @Gudahtt

**Review by Gudahtt — DISMISSED** [review]
LGTM!

**Comment by Gudahtt** [comment]
It just occurred to me - is #521 breaking? i.e. will mobile changes be required with that update, or will it "just work"? cc @rickycodes

**Review by Gudahtt — APPROVED** [review]
I've just looked over #521 and it doesn't appear to be breaking to me, so I'll re-approve. But I'd appreciate a second opinion on that. If it does require changes, it should be easy enough to create a v14 and copy over these changelog entries.

**Comment by rickycodes** [comment]
@Gudahtt that change is non breaking and is backwards compatible.

**Comment by adonesky1** [comment]
I'm going to pull #536 into this release.

**Comment by Gudahtt** [comment]
Thanks @rickycodes!

I can re-approve when #536 has been added to the changelog

**Comment by rickycodes** [comment]
Actually, I just took a closer look and forgot about: https://github.com/MetaMask/controllers/pull/521/files#diff-b062adcd7492a9ed99638930334523485e5146cf42195fa01bd4cddd7df90491R503-L500 (which does appear to be a breaking change). my bad!

**Comment by Gudahtt** [comment]
It seems like we could revert that one line deletion to keep this a non-breaking release :thinking: It doesn't seem essential to me, since `gasPrice` is deleted later on before signing or submitting on EIP-1559 networks.

**Comment by adonesky1** [comment]
>It seems like we could revert that one line deletion to keep this a non-breaking release 🤔 It doesn't seem essential to me, >since gasPrice is deleted later on before signing or submitting on EIP-1559 networks.
@rickycodes thoughts on this ^?

**Comment by andrepimenta** [comment]
> It seems like we could revert that one line deletion to keep this a non-breaking release 🤔 It doesn't seem essential to me, since `gasPrice` is deleted later on before signing or submitting on EIP-1559 networks.

@Gudahtt So the problem was that when a dapp did not suggest a gasPrice, we were adding a gasPrice and in that case the UI didn't have any way of telling if the gasPrice on the transaction object was suggested from the dapp or added by us.
If we remove it everything still works because we have QAd, so I guess we can remove it?

**Comment by adonesky1** [comment]
>@Gudahtt So the problem was that when a dapp did not suggest a gasPrice, we were adding a gasPrice. And in that case the UI didn't have any way of telling if the gasPrice on the transaction object was suggested from the dapp or added by us.
We have QAd and everything is working.

@andrepimenta are you saying this isn't in fact a breaking change?

**Comment by adonesky1** [comment]
@Gudahtt ok just spoke with @andrepimenta and @rickycodes and it seems that we should bump this to a major version release and indicate that this is a breaking change for the reason that Andre pointed out above. We shouldn't be adding a gas price here. Consumers should check if a gas price is included in the transaction object themselves and calculate and add it if need be.

**Comment by adonesky1** [comment]
@andrepimenta would you mind looking over the changelog line related to the breaking change?

**Review by danjm — APPROVED** [review]

**Comment by adonesky1** [comment]
@Gudahtt looks like this publish flow didn't work -- probably because the branch name didn't match the release version after the bump? How best to resolve this tangle?

**Comment by wachunei** [comment]
afaik publish is manual

**Comment by Gudahtt** [comment]
The GitHub release publishing and tagging should have been automated; that was indeed caused by the version change. In the future this would be easier to manage by closing the PR and creating a new one, but it's relatively easy to fix by publishing it manually as well

**Comment by Gudahtt** [comment]
>A new version of the package @metamask/controllers (14.0.0) was published at 2021-07-27T21:35:36.031Z from
[redacted]. The shasum of this package was 7621acd14d2fc007a6c6f4e3ba00022390914cea.

**Pull request merged**
